compiler/z/codegen/OMRCodeGenerator.cpp

```diff
@@ -6496,33 +6467,8 @@ OMR::Z::CodeGenerator::doBinaryEncoding()
 
       self()->setLabelHashTable(notPrintLabelHashTable);
       }
-   else
-      { //sets values for warmCodeEnd and coldCodeStart to avoid failing later assertions
-        //that warmcode length < estimated warmcode length
-      if (data.cursorInstruction == data.preProcInstruction)
-         {
-         self()->setPrePrologueSize(self()->getBinaryBufferCursor() - self()->getBinaryBufferStart());
-         }
-      if (self()->allowSplitWarmAndColdBlocks())
-         {
-         self()->setWarmCodeEnd(self()->getBinaryBufferCursor());
-         }
-      self()->setColdCodeStart(coldCode);
-      self()->setBinaryBufferCursor(coldCode);
-      }
-
-   if ( !self()->allowSplitWarmAndColdBlocks() )
-      {
-      // We need this, otherwise getWarmCodeLength() would compute a -ive value,
-      // which in-turn would make the later assert check fail. The negative
-      // value comes about because getCodeStart() is larger _binaryBufferStart,
-      // which is what warmCodeEnd is initlaized to.
-      self()->setWarmCodeEnd(self()->getCodeStart());
-      }
-
 
```

Comment:
On second thought, can we get rid off code setting ```doBinaryEncoding``` to true and the ```if``` statement checking if ```doBinaryEncoding``` is true (if that is the only live code path)?